### PR TITLE
Fix handling of `analyze_channels` for EEG data, where some steps (like ERP calculation) would previously fail

### PR DESCRIPTION
**Pull request by hoechenberger**
This was reported at https://mne.discourse.group/t/error-when-setting-analyze-channels-using-mne-bids-pipeline

### Before merging …

- [x] Changelog has been updated (`docs/source/changes.md`)

### CODE REVIEW

**Comment by larsoner** [comment]
For stuff like this as much as possible I still try to follow TDD and modify an example/test config to show the problem. Maybe one of the `ERP_CORE` datasets:
```
$ git grep "eeg_reference = "
...
mne_bids_pipeline/tests/configs/config_ERP_CORE.py:    eeg_reference = ["P9", "P10"]
mne_bids_pipeline/tests/configs/config_ERP_CORE.py:    eeg_reference = ["P9", "P10"]
mne_bids_pipeline/tests/configs/config_ERP_CORE.py:    eeg_reference = ["P9", "P10"]
mne_bids_pipeline/tests/configs/config_ERP_CORE.py:    eeg_reference = ["P9", "P10"]
mne_bids_pipeline/tests/configs/config_ERP_CORE.py:    eeg_reference = ["P9", "P10"]
mne_bids_pipeline/tests/configs/config_ERP_CORE.py:    eeg_reference = "average"
mne_bids_pipeline/tests/configs/config_ERP_CORE.py:    eeg_reference = ["P9", "P10"]
...
```
And it looks like `analyze_channels` is completely untested!
```
$ git grep "analyze_channels = "
mne_bids_pipeline/_config.py:    analyze_channels = ['Pz']
mne_bids_pipeline/_config_utils.py:        analyze_channels = cfg.analyze_channels
mne_bids_pipeline/_config_utils.py:            analyze_channels = cfg.ch_types
```
So we should probably add it here. @hoechenberger feel free to try it on `main` to ensure it fails then add to this PR, or I can do it.

**Comment by hoechenberger** [comment]
@larsoner Yes absolutely, I forgot to mention that I was actually planning to add a test case

I will do that when I have time; changing to a draft PR now!

File: mne_bids_pipeline/_config_utils.py [hunk 1]
```diff
@@ -423,22 +423,21 @@ def get_mf_ctc_fname(
 def _restrict_analyze_channels(
     inst: RawEpochsEvokedT, cfg: SimpleNamespace
 ) -> RawEpochsEvokedT:
-    if cfg.analyze_channels:
```
Comment (hoechenberger) [inline mne_bids_pipeline/_config_utils.py:426]: This should always be true with valid configuration options

File: mne_bids_pipeline/tests/configs/config_ERP_CORE.py [hunk 2]
```diff
@@ -224,6 +259,41 @@
     }
 
     eeg_reference = "average"
+    # Analyze all EEG channels -- we only specify the channels here for the purpose of
+    # demonstration
+    analyze_channels = [
```
Comment (hoechenberger) [inline mne_bids_pipeline/tests/configs/config_ERP_CORE.py:264]: test with average reference

File: mne_bids_pipeline/tests/configs/config_ERP_CORE.py [hunk 3]
```diff
@@ -212,6 +212,41 @@
     }
 
     eeg_reference = ["P9", "P10"]
+    # Analyze all EEG channels -- we only specify the channels here for the purpose of
+    # demonstration
+    analyze_channels = [
```
Comment (hoechenberger) [inline mne_bids_pipeline/tests/configs/config_ERP_CORE.py:217]: test with custom reference

File: mne_bids_pipeline/_config_utils.py (hunk 1 above)
Comment (larsoner) [reply mne_bids_pipeline/_config_utils.py:426]: True but to save us some pain you should add a `Len` or similar annotation to this to make sure it is non-empty

https://github.com/mne-tools/mne-bids-pipeline/blob/b52742e0b1a08deac963f59dafff4ad2d0578a62/mne_bids_pipeline/_config.py#L385

Comment (larsoner) [reply mne_bids_pipeline/_config_utils.py:426]: (then `pydantic` will throw an error if it's an empty list)

**Comment by hoechenberger** [comment]
@larsoner I'm trying to run the tests locally and it appears that auto-downloading the test datasets doesn't work? Or am I doing something wrong?

```python
❯ pytest -k ERP_CORE_N170 --download
======================================= test session starts =======================================
platform darwin -- Python 3.11.5, pytest-7.4.3, pluggy-1.3.0 -- /Users/richardhochenberger/mambaforge/envs/mne/bin/python3.11
cachedir: .pytest_cache
PyQt6 6.6.0 -- Qt runtime 6.6.0 -- Qt compiled 6.6.0
rootdir: /Users/richardhochenberger/Development-new/mne-bids-pipeline
configfile: pyproject.toml
testpaths: mne_bids_pipeline
plugins: harvest-1.10.4, timeout-2.2.0, qt-4.2.0, cov-4.1.0, anyio-4.1.0, typeguard-4.1.5
collected 29 items / 28 deselected / 1 selected

mne_bids_pipeline/tests/test_run.py::test_run[ERP_CORE_N170] Downloading ERP_CORE
--------------------
Dataset directory already exists; remove it if you wish to re-download the dataset


┌────────┬ Welcome aboard MNE-BIDS-Pipeline! 👋 ───────────────────────────────────────────────────
│17:23:17│ 📝 Using configuration: /Users/richardhochenberger/Development-new/mne-bids-pipeline/mne_bids_pipeline/tests/configs/config_ERP_CORE.py
│17:23:17│ ❌ Overriding config.task = 'N170'
│17:23:17│ 👾 Dask initializing with 4 workers …
│17:23:17│ 📂 Dask temporary directory: /Users/richardhochenberger/mne_data/derivatives/mne-bids-pipeline/ERP_CORE/.dask-worker-space
│17:23:19│ 🌎 Dask client dashboard: http://127.0.0.1:8787/status
└────────┴
┌────────┬ init/_01_init_derivatives_dir ──────────────────────────────────────────────────────────
│17:23:19│ ✅ Output directories already exist …
└────────┴ done (1s)
┌────────┬ init/_02_find_empty_room ───────────────────────────────────────────────────────────────
│17:23:19│ ⏩ Skipping, empty-room data only relevant for MEG …
└────────┴ done (1s)
┌────────┬ preprocessing/_01_data_quality ─────────────────────────────────────────────────────────
FAILED                         [100%]

============================================ FAILURES =============================================
_____________________________________ test_run[ERP_CORE_N170] _____________________________________
../../mambaforge/envs/mne/lib/python3.11/site-packages/joblib/parallel.py:1423: in dispatch_one_batch
    tasks = self._ready_batches.get(block=False)
../../mambaforge/envs/mne/lib/python3.11/queue.py:168: in get
    raise Empty
E   _queue.Empty

During handling of the above exception, another exception occurred:
mne_bids_pipeline/tests/test_run.py:206: in test_run
    main()
mne_bids_pipeline/_main.py:234: in main
    step_module.main(config=config_imported)
mne_bids_pipeline/steps/preprocessing/_01_data_quality.py:319: in main
    logs = parallel(
../../mambaforge/envs/mne/lib/python3.11/site-packages/joblib/parallel.py:1950: in __call__
    next(output)
../../mambaforge/envs/mne/lib/python3.11/site-packages/joblib/parallel.py:1588: in _get_outputs
    self._start(iterator, pre_dispatch)
../../mambaforge/envs/mne/lib/python3.11/site-packages/joblib/parallel.py:1571: in _start
    if self.dispatch_one_batch(iterator):
../../mambaforge/envs/mne/lib/python3.11/site-packages/joblib/parallel.py:1434: in dispatch_one_batch
    islice = list(itertools.islice(iterator, big_batch_size))
mne_bids_pipeline/steps/preprocessing/_01_data_quality.py:330: in <genexpr>
    for run, task in get_runs_tasks(
mne_bids_pipeline/_config_utils.py:268: in get_runs_tasks
    runs.extend(get_runs(config=config, subject=subject))
mne_bids_pipeline/_config_utils.py:219: in get_runs
    subj_runs = get_runs_all_subjects(config=config)
mne_bids_pipeline/_config_utils.py:138: in get_runs_all_subjects
    _get_runs_all_subjects_cached(
mne_bids_pipeline/_config_utils.py:159: in _get_runs_all_subjects_cached
    valid_runs_subj = _get_entity_vals_cached(
mne_bids_pipeline/_config_utils.py:53: in _get_entity_vals_cached
    return mne_bids.get_entity_vals(*args, **kwargs)
<decorator-gen-286>:12: in get_entity_vals
    ???
../../mambaforge/envs/mne/lib/python3.11/site-packages/mne_bids/path.py:1998: in get_entity_vals
    root = _check_fname(
<decorator-gen-0>:12: in _check_fname
    ???
../../Library/CloudStorage/OneDrive-dsm-firmenich/Documents/Development/mne-python/mne/utils/check.py:279: in _check_fname
    raise FileNotFoundError(f'{name} does not exist: "{fname}"')
E   FileNotFoundError: Root directory does not exist: "/Users/richardhochenberger/mne_data/ERP_CORE/sub-015"
---------------------------------------- Captured log call ----------------------------------------
INFO     distributed.scheduler:scheduler.py:1697 State start
INFO     distributed.scheduler:scheduler.py:4029   Scheduler at:     tcp://127.0.0.1:58895
INFO     distributed.scheduler:scheduler.py:4044   dashboard at:  http://127.0.0.1:8787/status
INFO     distributed.scheduler:scheduler.py:7592 Registering Worker plugin shuffle
INFO     distributed.nanny:nanny.py:361         Start Nanny at: 'tcp://127.0.0.1:58898'
INFO     distributed.nanny:nanny.py:361         Start Nanny at: 'tcp://127.0.0.1:58900'
INFO     distributed.nanny:nanny.py:361         Start Nanny at: 'tcp://127.0.0.1:58902'
INFO     distributed.nanny:nanny.py:361         Start Nanny at: 'tcp://127.0.0.1:58904'
INFO     distributed.scheduler:scheduler.py:4381 Register worker <WorkerState 'tcp://127.0.0.1:58908', name: 1, status: init, memory: 0, processing: 0>
INFO     distributed.scheduler:scheduler.py:5791 Starting worker compute stream, tcp://127.0.0.1:58908
INFO     distributed.core:core.py:1018 Starting established connection to tcp://127.0.0.1:58913
INFO     distributed.scheduler:scheduler.py:4381 Register worker <WorkerState 'tcp://127.0.0.1:58907', name: 2, status: init, memory: 0, processing: 0>
INFO     distributed.scheduler:scheduler.py:5791 Starting worker compute stream, tcp://127.0.0.1:58907
INFO     distributed.core:core.py:1018 Starting established connection to tcp://127.0.0.1:58914
INFO     distributed.scheduler:scheduler.py:4381 Register worker <WorkerState 'tcp://127.0.0.1:58909', name: 3, status: init, memory: 0, processing: 0>
INFO     distributed.scheduler:scheduler.py:5791 Starting worker compute stream, tcp://127.0.0.1:58909
INFO     distributed.core:core.py:1018 Starting established connection to tcp://127.0.0.1:58918
INFO     distributed.scheduler:scheduler.py:4381 Register worker <WorkerState 'tcp://127.0.0.1:58910', name: 0, status: init, memory: 0, processing: 0>
INFO     distributed.scheduler:scheduler.py:5791 Starting worker compute stream, tcp://127.0.0.1:58910
INFO     distributed.core:core.py:1018 Starting established connection to tcp://127.0.0.1:58917
INFO     distributed.scheduler:scheduler.py:5549 Receive client connection: Client-mne-bids-pipeline-d59252b0-e08c-11ee-a095-0ad92e5817a9
INFO     distributed.core:core.py:1018 Starting established connection to tcp://127.0.0.1:58919
- generated xml file: /Users/richardhochenberger/Development-new/mne-bids-pipeline/junit-results.xml -
====================================== slowest 10 durations =======================================
2.68s call     mne_bids_pipeline/tests/test_run.py::test_run[ERP_CORE_N170]
0.01s setup    mne_bids_pipeline/tests/test_run.py::test_run[ERP_CORE_N170]
0.00s teardown mne_bids_pipeline/tests/test_run.py::test_run[ERP_CORE_N170]
===================================== short test summary info =====================================
FAILED mne_bids_pipeline/tests/test_run.py::test_run[ERP_CORE_N170] - FileNotFoundError: Root directory does not exist: "/Users/richardhochenberger/mne_data/ERP_COR...
================================ 1 failed, 28 deselected in 4.72s =================================
```

**Comment by hoechenberger** [comment]
@larsoner Both, N170 and N2pc, fail to run on `main` with the configuration settings used here (reproduces the bug); everything is green on this PR branch. This is ready for review & merge

File: mne_bids_pipeline/_config_utils.py (hunk 1 above)
Comment (larsoner) [reply mne_bids_pipeline/_config_utils.py:426]: Pushed `Annotated[..., MinLen(1)]`

**Comment by larsoner** [comment]
Marking for merge-when-green, thanks @hoechenberger !

**Comment by hoechenberger** [comment]
@larsoner Think we can cut a 1.7 release with this?

**Comment by larsoner** [comment]
Sure or a 1.6.1

**Comment by hoechenberger** [comment]
@larsoner I tagged a new release but the doc deployment is failing :( https://app.circleci.com/pipelines/github/mne-tools/mne-bids-pipeline/4328/workflows/045c8d95-d657-4020-9ecf-71c00f53e2d9/jobs/60266/parallel-runs/0/steps/0-109

**Comment by larsoner** [comment]
Looks like:
```
remote: fatal: pack exceeds maximum allowed size (2.00 GiB)        
```
one solution is to archive or just remove some old doc versions like 1.0, 1.1, etc.

**Comment by larsoner** [comment]
@hoechenberger WDYT about deleting 1.0 and 1.1?

**Comment by larsoner** [comment]
FYI to fix things I re-ran the failed step with SSH (though I guess I could have done it locally), then SSH'ed in and did:
```console
$ source $BASH_ENV
$ cd project
$ git checkout main  # was on gh-pages
$ mike list --config-file docs/mkdocs.yml
Generating option->step mapping: 100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████| 56/56 [00:00<00:00, 228.61it/s]
warning: gh-pages is unrelated to origin/gh-pages
dev
1.7 [stable]
1.6
1.5
1.4
1.3
1.2
1.1
1.0
$ mike delete 1.0 --config-file docs/mkdocs.yml --ignore-remote-status
... # same for 1.1, 1.2, 1.3, and 1.4
$ git checkout gh-pages
$ git reset $(git commit-tree HEAD^{tree} -m "Deploy and squash docs [ci skip]")
$ git push origin --force gh-pages
...
Writing objects: 100% (1041/1041), 1.88 GiB | 70.96 MiB/s, done.
...
 + 9988ef4...ad8b0d3 gh-pages -> gh-pages (forced update)
```
and now things look okay:

https://mne.tools/mne-bids-pipeline/stable/

`dev` might have gotten reset but that should be fixed the next time we merge, which will hopefully be soon enough.

Note that we do get this warning [in deployment](https://github.com/mne-tools/mne-bids-pipeline/actions/runs/8284922599/job/22671715021#step:2:12):
```
Warning: Uploaded artifact size of 2875160017 bytes exceeds the allowed size of 1 GB. Deployment might fail.
```
So we might have more compression / size reduction to figure out at some point

**Pull request merged**